### ISSUE TITLE
Android TalkBack screen reader doesn't tell the user which item is selected in a .NET MAUI CollectionView

### ISSUE THREAD

**Issue by gbarkerz** [dotnet/maui#21375]
### Description

When navigating the Android TalkBack screen reader through the items in a CollectionView, when TalkBack encounters the selected item, its announcement should begin with: "selected". For an example of this behavior, move TalkBack through the items in a Spotify list of tracks in an album. However, in a .NET 8.0 MAUI CollectionView, TalkBack does not say "selected" when it encounters the selected item. 

### Steps to Reproduce

1. Build the sample app at https://github.com/gbarkerz/MauiCollectionViewTalkBackTest/tree/master, and run it on an Android device.
2. Tap on an item in the CollectionView to select it.
3. Start the TalkBack screen reader on the device, and move it through the items in the CollectionView.

Result: TalkBack never says "selected" when encountering the selected item. 

Note that in the "Version with bug" section below, I've added the output of the "dotnet workload list". in Visual Studio, the project's properties say that the Target .NET Runtime is net8.0.

The image here shows TalkBack's caption when encountering the selected item in the test app.

<img width="236" alt="image" src="https://github.com/dotnet/maui/assets/77085891/0b8881ca-2ca9-4f8f-8289-7b710d1e468a">

### Link to public reproduction project repository

https://github.com/gbarkerz/MauiCollectionViewTalkBackTest/tree/master

### Version with bug

9.0.0-preview.1.9973

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 14

### Did you find any workaround?

No workaround found.

**Comment by ninachen03** [comment]
Can repro it at Android on the latest 17.10.0 Preview 3.0 [34721.187.main] with net9.
